hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -3112,6 +3112,22 @@ def map(self, f):
         assert isinstance(self._type, tndarray)
         return ndarray_map
 
+    @typecheck_method(uri=str)
+    def save(self, uri):
+        """Write out the NDArray to the given path as in .npy format. If the URI does not
+        end with ".npy" the file extension will be appended. This method reflects the numpy
+        `save` method. NDArrays saved with this method can be loaded into numpy using numpy
+        `load`.
```

Comment:
Might be helpful to add a simple example of saving an NDArray and reading it back in numpy.